modular_splurt/code/datums/interactions/lewd/_lewd.dm

```diff
@@ -784,15 +810,16 @@
 	var/list/hell = list(
 		" pushes [u_His] [pick(ass)] into \the <b>[target]</b>'s crotch, [pick(list("ripping a fat[pick(list("", " and [pick(stank)]"))] one", " and subsequently [pick(braps)]"))][pick(list("", ". [jiggle]"))]",
 		" and \the <b>[target]</b> can smell the [prob(50) ? pick(stank) : "flatulent"] gas fill the room ass it seeps in between \the <b>[target]</b>'s thighs!",
-		" [pick(braps)] into \the <b>[target]</b>'s [brap_catcher ? brap_catcher.name : "crotch"]"
+		" [pick(braps)] into \the <b>[target]</b>'s [genital_name]"
 	)
 
 	message = "<span class='lewd'>\The <b>[src]</b>[pick(hell)]</span>"
 	visible_message(message, ignored_mobs = get_unconsenting(unholy = TRUE))
 	playlewdinteractionsound(loc, pick(GLOB.brap_noises), 50, 1, -1, ignored_mobs = get_unconsenting(unholy = TRUE))
 	if(!target.is_fucking(src, CUM_TARGET_ANUS))
 		target.set_is_fucking(src, CUM_TARGET_ANUS, target.has_penis(REQUIRE_EXPOSED) ? target.getorganslot(ORGAN_SLOT_PENIS) : (target.has_vagina(REQUIRE_EXPOSED) ? target.getorganslot(ORGAN_SLOT_VAGINA) : null))
-	target.handle_post_sex(NORMAL_LUST, CUM_TARGET_ANUS, src)
+	if(target.can_penetrating_genital_cum())
+		target.handle_post_sex(NORMAL_LUST, CUM_TARGET_ANUS, src)
```

Comment:
This one also uses the same proc for vagina and penis so it's likely the same problem